src/likelihoods/negativebinomial.jl

```diff
@@ -1,36 +1,169 @@
+abstract type NBParam end
+
+abstract type NBParamProb <: NBParam end
+default_invlink(::NBParamProb) = logistic
+abstract type NBParamMean <: NBParam end
+default_invlink(::NBParamMean) = exp
+
 """
-    NegativeBinomialLikelihood(l=logistic; successes::Real=1)
+    NegativeBinomialLikelihood(param::NBParam, invlink::Union{Function,Link})
 
-Negative binomial likelihood with number of successes `successes`.
+There are many possible parametrizations for the Negative Binomial likelihood.
+We follow the convention laid out in p.137 of [^1] and provide some common parametrizations.
+The `NegativeBinomialLikelihood` has a special structure; the first type parameter `NBParam`
+defines in what parametrization the latent function is used, and contains the other (scalar) parameters.
+`NBParam` itself has two subtypes:
+- `NBParamProb` for parametrizations where `f -> p`, the probability of success of a Bernoulli event
+- `NBParamMean` for parametrizations where `f -> μ`, the expected number of events
 
-```math
-    p(k|successes, f) = \\frac{\\Gamma(k+successes)}{k! \\Gamma(successes)} l(f)^successes (1 - l(f))^k
+## `NBParam` predefined types
+
+### `NBParamProb` types with `p = invlink(f)` the probability of success
+- [`NBParamSuccess`](@ref): This is the definition used in [`Distributions.jl`](https://juliastats.org/Distributions.jl/latest/univariate/#Distributions.NegativeBinomial).
+- [`NBParamFailure`](@ref): This is the definition used in [Wikipedia](https://en.wikipedia.org/wiki/Negative_binomial_distribution).
+
+
+### `NBParamMean` types with `μ = invlink(f)` the mean/expected number of events
+- [`NBParamI`](@ref): Mean is linked to `f` and variance is given by `μ(1 + α)`
+- [`NBParamII`](@ref): Mean is linked to `f` and variance is given by `μ(1 + α * μ)`
+- [`NBParamPower`](@ref): Mean is linked to `f` and variance is given by `μ + α * μ^ρ`
```

Comment:
Should this be 
```suggestion
- [`NBParamPower`](@ref): Mean is linked to `f` and variance is given by `μ(1 + α * μ^ρ)`
```
?